stem/output.py

```diff
@@ -0,0 +1,219 @@
+        node_output (bool):
+        skin_output (bool):
+        plane_output (List[str]):
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        point_data_configuration (List[str]):
+    """
+
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+
+    gid_post_mode: str = "GiD_PostBinary"
+    file_label: str = "step"
+    output_control_type: str = "step"
+    output_interval: int = 1
+    body_output: bool = True
+    node_output: bool = False
+    skin_output: bool = False
+    # optional
+    plane_output: List[str] = field(default_factory=lambda: [])
+    point_data_configuration: List[str] = field(default_factory=lambda: [])
+
+    def validate(self):
+        pass
+
+
+@dataclass
+class VtkOutputParameters(OutputParametersABC):
+    """
+    Class containing the output parameters for GiD output
+
+    Attributes:
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        file_format (str): file format for VTK, either `binary` or `ascii` are allowed.
+        output_precision (int):
+        output_control_type (str): type of output control, either `step` or `time`.
+        output_interval (float):
+    """
+    # general inputs
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+    # VTK specif inputs
+    file_format: str = "binary"
+    output_precision: int = 7
+    output_control_type: str = "step"
+    output_interval: float = 1.0
+
+    def validate(self):
+        detect_tensor_outputs(requested_outputs=self.gauss_point_results)
+
+
+@dataclass
+class JsonOutputParameters(OutputParametersABC):
+    """
+    Class containing the output parameters for JSON output
+
+    Attributes:
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        time_frequency (float):
+        sub_model_part_name (str):
+    """
+    # general inputs
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+    # JSON specif inputs
+    time_frequency: float = 1.0
+    sub_model_part_name: str = ""
+
+    def validate(self):
+        detect_tensor_outputs(requested_outputs=self.gauss_point_results)
+
+
+class Output:
+    """
+    Class containing output information for postprocessing
+
+    Attributes:
+        part_name (str): name of the model part
+        output_path (str): output path for the output
+            - for GID, it represents the name (or relative path) ot the bin file.
+              For relative paths, the stem is referred as the name and the previous
+              as the directory. They will be created if they do not exist.
+              example1: `test1` results in the test1.bin file in the current folder.
+              example2: `path1/path2/test2` results in the test2.bin file in the
```

Comment:
.bin extension is only there for binary output, ascii output has a different extension